internal/pkg/cli/pipeline_init.go

```diff
@@ -556,18 +598,21 @@ func (o *initPipelineOpts) secretName() string {
 
 func (o *initPipelineOpts) pipelineName() (string, error) {
 	if o.provider == ghProviderName {
-		return fmt.Sprintf(fmtGHPipelineName, o.appName, o.githubOwner, o.repoName), nil
+		return fmt.Sprintf(fmtGHPipelineName, o.appName, o.repoOwner, o.repoName), nil
 	}
 	if o.provider == ccProviderName {
 		return fmt.Sprintf(fmtCCPipelineName, o.appName, o.repoName), nil
 	}
+	if o.provider == bbProviderName {
+		return fmt.Sprintf(fmtBBPipelineName, o.appName, o.repoOwner, o.repoName), nil
+	}
 	return "", fmt.Errorf("unable to create pipeline name for repo %s from provider %s", o.repoName, o.provider)
 }
 
 func (o *initPipelineOpts) pipelineProvider() (manifest.Provider, error) {
 	if o.provider == ghProviderName {
```

Comment:
since all these if-statements end with `return manifest.NewProvider` we can do this:
```go
var config interface{}
switch o.provider {
case ghProviderName:
  config = &manifest.GitHubProps{...}
case ccProvideName:
   config = &manifest.CodeCommitProps{...}
case bbProviderName:
   config = &manifest.BitBucketProps{...}
default:
   return nil, fmt.Errorf(...)
}
return manifest.NewConfig(config)
```

Reply:
Thanks. Done. And also did that for `pipelineName()`.